op-challenger/challenger/factory.go

```diff
@@ -0,0 +1,38 @@
+package challenger
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var ErrMissingFactoryEvent = errors.New("missing factory event")
+
+// BuildDisputeGameLogFilter creates a filter query for the DisputeGameFactory contract.
+//
+// The `DisputeGameCreated` event is encoded as:
+// 0: address indexed disputeProxy,
+// 1: GameType indexed gameType,
+// 2: Claim indexed rootClaim,
+func BuildDisputeGameLogFilter(contract *abi.ABI) (ethereum.FilterQuery, error) {
+	event := contract.Events["DisputeGameCreated"]
+
+	if event.ID == (common.Hash{}) {
+		return ethereum.FilterQuery{}, ErrMissingFactoryEvent
+	}
+
+	query := ethereum.FilterQuery{
+		Topics: [][]common.Hash{
+			{event.ID},
+		},
+	}
+
+	return query, nil
+}
+
+// BuildDisputeGameLogFilter creates a filter query for the DisputeGameFactory contract.
+func (c *Challenger) BuildDisputeGameLogFilter() (ethereum.FilterQuery, error) {
```

Comment:
Not sure this needs to be a method on `Challenger` given it just delegates to the standalone function.

Reply:
We need it to access the internal `c.dgfABI` field

Reply:
It's only called from the `NewFactorySubscription` method on `Challenger` though so that already has access to `dgfABI`.  I don't _think_ we need to export either of these methods from the module.